src/plugins/expressions/common/expression_functions/types.ts

```diff
@@ -94,3 +102,39 @@ export interface ExpressionFunctionDefinition<
  * Type to capture every possible expression function definition.
  */
 export type AnyExpressionFunctionDefinition = ExpressionFunctionDefinition<any, any, any, any>;
+
+/**
+ * Anybody registering an expression function should add a property to this
+ * interface so that modules importing `ExpressionFunctionDefinitions` can
+ * have the interfaces of all expression functions available to them.
+ *
+ * @example
+ *  declare module '../../plugins/expressions/public' {
+ *    interface ExpressionFunctionDefinitions {
+ *      myExprFunctionName: MyExpressionFunctionDefinition;
+ *    }
+ *  }
+ *
+ *  declare module '../../plugins/expressions/server' {
+ *    interface ExpressionFunctionDefinitions {
+ *      myExprFunctionName: MyExpressionFunctionDefinition;
+ *    }
+ *  }
+ *
+ * @public
+ */
+export interface ExpressionFunctionDefinitions {
```

Comment:
I couldn't decide on the name to use here. `ExpressionFunctionDefinitions` feels most accurate, but I have noticed elsewhere (like ui_actions) we are using a `*Mapping` naming convention for interfaces like these which are augmented by other plugins. `ExpressionFunctionMapping`? Open to suggestions.